modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java

```diff
@@ -344,7 +342,160 @@ public void unShareB2BApplication() throws JSONException {
+                .body(requestBody)
+                .when()
+                .post(endpointURL);
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_CREATED);
+    }
+
+    @Test(groups = "discoveryTests", dependsOnMethods = "testAddDiscoveryAttributesToOrganization")
+    public void testGetDiscoveryAttributesOfOrganizations() {
+
+        String endpointURL = ORGANIZATION_MANAGEMENT_API_BASE_PATH + ORGANIZATION_DISCOVERY_API_PATH;
+        Response response = given().auth().preemptive().oauth2(m2mToken)
+                .contentType(ContentType.JSON)
+                .when()
+                .get(endpointURL);
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_OK)
+                .body("totalResults", equalTo(1))
+                .body("organizations[0].organizationId", equalTo(organizationID))
+                .body("organizations[0].organizationName", equalTo("Greater Hospital"))
+                .body("organizations[0].attributes[0].type", equalTo("emailDomain"))
+                .body("organizations[0].attributes[0].values[0]", equalTo("abc.com"));
+    }
+
+    @Test(groups = "discoveryTests", dependsOnMethods = "testGetDiscoveryAttributesOfOrganizations")
+    public void testGetDiscoveryAttributesOfOrganization() {
+
+        String endpointURL = ORGANIZATION_MANAGEMENT_API_BASE_PATH + PATH_SEPARATOR + organizationID
+                                + ORGANIZATION_DISCOVERY_API_PATH;
+        Response response = given().auth().preemptive().oauth2(m2mToken)
+                .contentType(ContentType.JSON)
+                .when()
+                .get(endpointURL);
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_OK)
+                .body("attributes[0].type", equalTo("emailDomain"))
+                .body("attributes[0].values[0]", equalTo("abc.com"));
+    }
+
+    @Test(groups = "discoveryTests", dependsOnMethods = "testGetDiscoveryAttributesOfOrganization")
+    public void testUpdateDiscoveryAttributesOfOrganization() throws IOException {
+
+        String endpointURL = ORGANIZATION_MANAGEMENT_API_BASE_PATH + PATH_SEPARATOR + organizationID
+                                + ORGANIZATION_DISCOVERY_API_PATH;
+        String requestBody = readResource("update-discovery-config-request-body.json");
+        Response response = given().auth().preemptive().oauth2(m2mToken)
+                .contentType(ContentType.JSON)
+                .body(requestBody)
+                .when()
+                .put(endpointURL);
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_OK)
+                .body("attributes[0].values[0]", equalTo("xyz.com"));
```

Comment:
shall we test the update with multiple email domains

Reply:
The update method is now tested with 3 domains